binder/src/main/scala/PanamaCIRCT.scala

```diff
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.internal.panama.circt
+
+import java.lang.foreign._
+import java.lang.foreign.MemorySegment.NULL
+import java.lang.foreign.ValueLayout._
+
+import org.llvm.circt
+import org.llvm.circt.CAPI
+
+// Wrapper for CIRCT APIs with Panama framework
+class PanamaCIRCT {
+  // Open an arena for memory management of MLIR API calling in this context instance
+  private val arena = Arena.openConfined()
+
+  // Create MLIR context and register dialects we need
+  private val mlirCtx = {
+    val mlirCtx = CAPI.mlirContextCreate(arena)
+
+    // Register dialects
+    Seq(
+      CAPI.mlirGetDialectHandle__firrtl__(arena),
+      CAPI.mlirGetDialectHandle__chirrtl__(arena)
```

Comment:
By default these two dialect must be registered.
But if emitting Verilog and use full feature of CIRCT need more things to add.
See dialect, firtool loaded [here](https://github.com/llvm/circt/blob/08778859fc22dc1b5625b310ee18e66cca86e0e1/tools/firtool/firtool.cpp#L573-L576)
It can be live in the future work